File: vacuum-cleaner-route/vacuum_cleaner_route.go
package vacuumcleanerroute

func IsCompleteRoute(moves string) bool {
	var (
		x int
		y int
	)
	for _, c := range moves {
		switch c {
		case 'U':
			y++
		case 'D':
			y--
		case 'R':
			x++
		case 'L':
			x--
		}
	}
	return x == 0 && y == 0
}

File: week-01/vacuum-cleaner-route/vacuum_cleaner_route.py
def is_complete_route(moves):
    x, y = 0, 0
    for move in moves:
        if move == 'U':
            y += 1
        if move == 'D':
            y -= 1
        if move == 'R':
            x += 1
        if move == 'L':
            x -= 1
    return x == 0 and y == 0
